libraries/standard/http/src/http_client.c

```diff
@@ -245,6 +245,986 @@ static int findHeaderValueParserCallback( http_parser * pHttpParser,
+ *
+ * @param[in] pParsingContext Parsing state containing information to notify
+ * the application of a complete header.
+ */
+static void processCompleteHeader( HTTPParsingContext_t * pParsingContext );
+
+/**
+ * @brief When parsing is complete an error could be indicated in
+ * pHttpParser->http_errno. This function translates that error into a library
+ * specific error code.
+ *
+ * @param[in] pHttpParser Third-party HTTP parsing context.
+ *
+ * @return One of the following:
+ * - #HTTP_SUCCESS
+ * - #HTTP_SECURITY_ALERT_RESPONSE_HEADERS_SIZE_LIMIT_EXCEEDED
+ * - #HTTP_SECURITY_ALERT_EXTRANEOUS_RESPONSE_DATA
+ * - #HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_CHUNK_HEADER
+ * - #HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_PROTOCOL_VERSION
+ * - #HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_STATUS_CODE
+ * - #HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_CHARACTER
+ * - #HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_CONTENT_LENGTH
+ * - #HTTP_PARSER_INTERNAL_ERROR
+ */
+static HTTPStatus_t processHttpParserError( http_parser * pHttpParser );
+
+/*-----------------------------------------------------------*/
+
+static void processCompleteHeader( HTTPParsingContext_t * pParsingContext )
+{
+    HTTPResponse_t * pResponse = NULL;
+
+    assert( pParsingContext != NULL );
+    assert( pParsingContext->pResponse != NULL );
+
+    pResponse = pParsingContext->pResponse;
+
+    /* A header is complete when both the last header field and value have been
+     * filled in. */
+    if( ( pParsingContext->pLastHeaderField != NULL ) &&
+        ( pParsingContext->pLastHeaderValue != NULL ) )
+    {
+        /* Increase the header count. */
+        pResponse->headerCount++;
+
+        LogDebug( ( "Response parsing: Found complete header: "
+                    "HeaderField=%.*s, HeaderValue=%.*s",
+                    pParsingContext->lastHeaderFieldLen,
+                    pParsingContext->pLastHeaderField,
+                    pParsingContext->lastHeaderValueLen,
+                    pParsingContext->pLastHeaderValue ) );
+
+        /* If the application registered a callback, then it must be notified. */
+        if( pResponse->pHeaderParsingCallback != NULL )
+        {
+            pResponse->pHeaderParsingCallback->onHeaderCallback(
+                pResponse->pHeaderParsingCallback->pContext,
+                ( const uint8_t * ) pParsingContext->pLastHeaderField,
+                pParsingContext->lastHeaderFieldLen,
+                ( const uint8_t * ) pParsingContext->pLastHeaderValue,
+                pParsingContext->lastHeaderValueLen,
+                pResponse->statusCode );
+        }
+
+        /* Prepare the next header field and value for the first invocation of
+         * httpParserOnHeaderFieldCallback() and
+         * httpParserOnHeaderValueCallback(). */
+        pParsingContext->pLastHeaderField = NULL;
+        pParsingContext->lastHeaderFieldLen = 0;
```

Comment:
0U?

Reply:
yes I will ad this for consistency. for some reason misra did not flag this.